addons/mail/static/src/discuss/call/common/rtc_service.js

```diff
@@ -1445,14 +1579,23 @@ export class Rtc {
             return;
         }
         const transceivers = rtcSession.peerConnection.getTransceivers();
-        const transceiver = transceivers[ORDERED_TRANSCEIVER_NAMES.indexOf("video")];
-        if (!transceiver) {
+        const cameraTransceiver = transceivers[ORDERED_TRANSCEIVER_NAMES.indexOf("camera")];
+        const screenTransceiver = transceivers[ORDERED_TRANSCEIVER_NAMES.indexOf("screen")];
+        if (!cameraTransceiver && !screenTransceiver) {
             return;
         }
-        transceiver.direction = this.getTransceiverDirection(
-            rtcSession,
-            Boolean(this.state.videoTrack)
-        );
+        if (cameraTransceiver) {
+            cameraTransceiver.direction = this.getTransceiverDirection(
+                rtcSession,
+                Boolean(this.state.cameraTrack)
+            );
+        }
+        if (screenTransceiver) {
+            screenTransceiver.direction = this.getTransceiverDirection(
+                rtcSession,
+                Boolean(this.state.screenTrack)
+            );
+        }
     }
```

Comment:
It works as long as we make the assumption that on the functional side, both streams are always displayed together

`rtcSession.videoComponentCount === 0` means no stream
`rtcSession.videoComponentCount !== 0` is always equal to `rtcSession.videoComponentCount >= 2` and means both streams.

Reply:
Yes I mean in terms of local transceiver direction:

The case you mention will be not be handled by this function, it is the remote (peer) who will have a remote `direction` `recvonly`/`inactive` and it will automatically result into a `currentDirection` that is appropriate)